packages/snaps-rpc-methods/src/permitted/requestPermissions.ts

```diff
@@ -0,0 +1,103 @@
+import type { JsonRpcEngineEndCallback } from '@metamask/json-rpc-engine';
+import type {
+  PermissionConstraint,
+  PermittedHandlerExport,
+  SubjectPermissions,
+} from '@metamask/permission-controller';
+import type {
+  RequestPermissionParams,
+  RequestPermissionsResult,
+} from '@metamask/snaps-sdk';
+import type { Snap, SnapPermissions } from '@metamask/snaps-utils';
+import type { JsonRpcRequest, PendingJsonRpcResponse } from '@metamask/utils';
+import deepEqual from 'fast-deep-equal';
+
+import type { MethodHooksObject } from '../utils';
+
+export type RequestPermissionsHooks = {
+  getSnap: () => Snap | undefined;
+  /**
+   * @returns Requested permission result.
+   */
+  requestPermissions: (
+    dynamicPermissions: RequestPermissionParams,
+  ) => Promise<SubjectPermissions<PermissionConstraint>>;
+};
+
+/**
+ * The `snap_requestPermissions` method implementation.
+ * Requests dynamic permissions at the runtime.
+ *
+ *
+ * @param request - The JSON-RPC request object.
+ * @param response - The JSON-RPC response object.
+ * @param _next - The `json-rpc-engine` "next" callback. Not used by this
+ * function.
+ * @param end - The `json-rpc-engine` "end" callback.
+ * @param hooks - The RPC method hooks.
+ * @param hooks.getSnap - Get Snap hook.
+ * @param hooks.requestPermissions - Request permission hook.
+ */
+async function requestPermissionsImplementation(
+  request: JsonRpcRequest<RequestPermissionParams>,
+  response: PendingJsonRpcResponse<RequestPermissionsResult>,
+  _next: unknown,
+  end: JsonRpcEngineEndCallback,
+  { getSnap, requestPermissions }: RequestPermissionsHooks,
+): Promise<void> {
+  const snap = getSnap();
+
+  const requestedPermissions = request.params as SnapPermissions;
+  const dynamicPermissions = snap?.manifest.dynamicPermissions;
+
+  if (!dynamicPermissions) {
+    throw new Error('This Snap have no dynamic permissions specified.');
+  }
+
+  const requestedPermissionsKeys = Object.keys(
+    requestedPermissions,
+  ) as (keyof typeof dynamicPermissions)[];
+
+  requestedPermissionsKeys.forEach((permissionKey) => {
+    if (!(permissionKey in dynamicPermissions)) {
```

Comment:
Can we use `hasProperty`?

Reply:
Refactored, using `hasProperty` from `utils` now.